realkd/subgroups.py

```diff
@@ -33,20 +34,21 @@ class Impact:
     Sex==female
     """
 
-    def __init__(self, data, target):
+    def __init__(self, data, target, labels=None):
+        data = validate_data(data, labels)
         self.m = len(data)
-        self.data = data.sort_values(target, ascending=False)  # data
+        self.data = data[argsort(data[target])[::-1]]
         self.data.reset_index(drop=True, inplace=True)
         self.target = target
         self.mean = self.data[self.target].mean()
 
     def __call__(self, q):
-        extent = self.data.loc[q]
+        extent = self.data[q]
```

Comment:
NOTE:
This is called both with a query (callable) and a list of extents, hence why the q(self.data) pattern was not used here and also why numpy input is not supported for Impact

Reply:
Ok, this is interesting. It means that one functionality of pandas that was used internally is indexing with a callable. Can we come up with a general refactoring for these cases? I suppose the straightforward solution is to have 

```
self.data[q(self.data)]
```

Does that look good to you?

Reply:
I don't believe that will work out of the box, since q could not be a callable.
A solution could be:
```
if callable(q):
  self.data[q(self.data)]
else:
  self.data[q]
```